consensus/core/src/context.rs

```diff
@@ -41,4 +44,47 @@ impl Context {
             metrics,
         }
     }
+
+    #[cfg(test)]
+    pub(crate) fn new_for_test() -> Self {
+        let (committee, _) = Committee::new_for_test(0, vec![1, 1, 1, 1]);
+        let metrics = test_metrics();
+        Context::new(
+            AuthorityIndex::new_for_test(0),
+            committee,
+            Parameters::default(),
+            ProtocolConfig::get_for_min_version(),
+            metrics,
+        )
+    }
+
+    #[cfg(test)]
```

Comment:
Marked them as `test` only for now as we probably don't want to modify the Context after creation for prod purposes, we can remove if needed later.

Reply:
Yeah we shouldn't need to modify `Context` for non test logic. But definitely the burden of using `Context` is in testing, where we need flexibility to construct and modify it.